python/cuml/cuml/accel/__main__.py

```diff
@@ -13,128 +13,206 @@
-    if convert_to_sklearn:
+    if ns.convert_to_sklearn:
+        with open(ns.convert_to_sklearn, "rb") as f:
+            if ns.format == "pickle":
+                import pickle
 
-        with open(convert_to_sklearn, "rb") as f:
-            if format == "pickle":
                 serializer = pickle
-            elif format == "joblib":
-                serializer = joblib
-            accelerated_estimator = serializer.load(f)
+            elif ns.format == "joblib":
+                import joblib
 
-        sklearn_estimator = accelerated_estimator.as_sklearn()
+                serializer = joblib
+            estimator = serializer.load(f)
 
-        with open(output, "wb") as f:
-            serializer.dump(sklearn_estimator, f)
+        # Conversion is only necessary for estimators built on `ProxyMixin`,
+        # estimators built with `ProxyBase` pickle transparently as their
+        # non-accelerated versions.
+        if isinstance(estimator, ProxyMixin):
+            estimator = estimator.as_sklearn()
 
+        with open(ns.output, "wb") as f:
+            serializer.dump(estimator, f)
         sys.exit()
 
-    if module:
-        (module,) = module
-        # run the module passing the remaining arguments
-        # as if it were run with python -m <module> <args>
-        sys.argv[:] = [module, *args.args]  # not thread safe?
-        runpy.run_module(module, run_name="__main__")
-    elif len(args) >= 1:
-        # Remove ourself from argv and continue
-        sys.argv[:] = args
-        runpy.run_path(args[0], run_name="__main__")
-    else:
-        if sys.stdin.isatty():
+    # Enable cudf.pandas if requested
+    if ns.cudf_pandas:
+        import cudf.pandas
+
+        cudf.pandas.install()
+
+    # Parse verbose into log_level
+    default_logger_level_index = list(logger.level_enum).index(
+        logger.level_enum.warn
+    )
+    log_level = list(logger.level_enum)[
+        max(0, default_logger_level_index - ns.verbose)
+    ]
+
+    # Enable acceleration
+    install(disable_uvm=ns.disable_uvm, log_level=log_level)
+
+    if ns.module is not None:
+        # Execute a module
+        sys.argv[:] = [ns.module, *ns.args]
+        runpy.run_module(ns.module, run_name="__main__")
+    elif ns.script is not None:
+        # Execute a script
+        sys.argv[:] = [ns.script, *ns.args]
+        runpy.run_path(ns.script, run_name="__main__")
+    elif sys.stdin.isatty():
+        # Start an interpreter as similar to `python` as possible
+        if sys.flags.quiet:
+            banner = ""
+        else:
             banner = f"Python {sys.version} on {sys.platform}"
-            site_import = not sys.flags.no_site
-            if site_import:
+            if not sys.flags.no_site:
                 cprt = 'Type "help", "copyright", "credits" or "license" for more information.'
                 banner += "\n" + cprt
-        else:
-            # Don't show prompts or banners if stdin is not a TTY
```

Comment:
We can't use `code` for non-interactive interpreters (e.g. not a TTY) since it swallows errors. This led to a ton of confusion during testing since failing test cases were accidentally passing since the errors didn't result in a failing returncode.

Using `execute_source` for `stdin` now results in stdout and returncodes matching the std python interpreter much better.